workflows/nallo.nf

```diff
@@ -0,0 +1,301 @@
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT LOCAL SUBWORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+include { PREPARE_GENOME             } from '../subworkflows/local/prepare_genome'
+include { BAM_TO_FASTQ               } from '../subworkflows/local/bam_to_fastq'
+include { ASSEMBLY                   } from '../subworkflows/local/genome_assembly'
+include { ASSEMBLY_VARIANT_CALLING   } from '../subworkflows/local/assembly_variant_calling'
+include { ALIGN_READS                } from '../subworkflows/local/align_reads'
+include { QC_ALIGNED_READS           } from '../subworkflows/local/qc_aligned_reads'
+include { STRUCTURAL_VARIANT_CALLING } from '../subworkflows/local/structural_variant_calling'
+include { SHORT_VARIANT_CALLING      } from '../subworkflows/local/short_variant_calling'
+include { CNV                        } from '../subworkflows/local/cnv'
+include { REPEAT_ANALYSIS            } from '../subworkflows/local/repeat_analysis'
+include { METHYLATION                } from '../subworkflows/local/methylation'
+include { PHASING                    } from '../subworkflows/local/phasing'
+include { SNV_ANNOTATION             } from '../subworkflows/local/snv_annotation'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT LOCAL/NF-CORE MODULES
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// local
+include { FQCRS                  } from '../modules/local/fqcrs'
+include { CONVERT_ONT_READ_NAMES } from '../modules/local/convert_ont_read_names'
+include { BUILD_INTERVALS        } from '../modules/local/build_intervals/main'
+include { SPLIT_BED_CHUNKS       } from '../modules/local/split_bed_chunks/main'
+
+// nf-core
+include { FASTQC                 } from '../modules/nf-core/fastqc/main'
+include { MULTIQC                } from '../modules/nf-core/multiqc/main'
+include { paramsSummaryMap       } from 'plugin/nf-validation'
+include { paramsSummaryMultiqc   } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { softwareVersionsToYAML } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { methodsDescriptionText } from '../subworkflows/local/utils_nfcore_nallo_pipeline'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+workflow NALLO {
+
+    take:
+    ch_input
+
+    main:
+    ch_versions      = Channel.empty()
+    ch_multiqc_files = Channel.empty()
+
+    // Mandatory input files
+    ch_fasta          = Channel.fromPath(params.fasta).map { it -> [it.simpleName, it] }.collect()
+
+    // Optional input files
+    ch_extra_snfs      = params.extra_snfs      ? Channel.fromSamplesheet('extra_snfs' , immutable_meta: false)
```

Comment:
Curious about `immutable_meta: false` as I haven't seen it before. Why is it needed?

Reply:
I think fromSamplesheet will create an immutable object, and might have been that I wanted to change the meta of these at some point. Tested it manually and it works without.